module_utils/network_lsr/nm/connection.py

```diff
@@ -0,0 +1,72 @@
+# SPDX-License-Identifier: BSD-3-Clause
+
+# Handle NM.RemoteConnection
+
+import logging
+
+# Relative import is not support by ansible 2.8 yet
+# pylint: disable=import-error, no-name-in-module
+from ansible.module_utils.network_lsr.nm.client import get_client  # noqa:E501
+from ansible.module_utils.network_lsr.nm.client import get_mainloop  # noqa:E501
+from ansible.module_utils.network_lsr.nm.active_connection import (  # noqa:E501
+    deactivate_active_connection,
+)
+from ansible.module_utils.network_lsr.nm.error import LsrNetworkNmError  # noqa:E501
+
+# pylint: enable=import-error, no-name-in-module
+
+
+def delete_remote_connection(nm_profile, timeout, check_mode):
+    if not nm_profile:
+        logging.info("NULL NM.RemoteConnection, no need to delete")
+        return False
+
+    # Whether profile has to be deactivated first
+    nm_client = get_client()
+    for nm_ac in nm_client.get_active_connections():
+        cur_nm_profile = nm_ac.get_connection()
+        if cur_nm_profile and cur_nm_profile.get_uuid() == nm_profile.get_uuid():
+            deactivate_active_connection(nm_ac, timeout, check_mode)
```

Comment:
this branch does something quite different from before. Before, the code made an effort to make the profile only volatile (if the API supports that). The effect is that the profile only gets deleted after the profile is no longer active.

This branch, explicitly deactivates the profile (if active) and removes it right away.

Sure, the way how we delete a profile could also be differently. But it was done that way for reasons. The commit message makes no mention why this changes and why the new way is preferable.

I also wonder why the profile gets first deactivated, only to delete it afterwards. That does not seem necessary. There is a code comment `"Whether profile has to be deactivated first"`, but it would be good if the code comment explains *why* this is done.

Reply:
OK. I missed this use case. I will restored back to old behavior.